### ISSUE TITLE
[Java] : (New rule) Lossy conversion needs to be flagged in Java code

### ISSUE THREAD

**Issue by Fernal73** [pmd/pmd#2023]
**Affects PMD Version:** 
6.17.0

**Rule:**
New rule.

**Description:**
See title.
[Best explained via this tweet.](https://twitter.com/heinzkabutz/status/1174340783073366017) 

**Code Sample demonstrating the issue:**
```
package javapuzzles;

public enum Addition {
  ;

  public static void main(String... args) {
    int a = 45;
    int d;
    double b = 3.1415987;
    long c = 70000000;
    a += b; // not flagged
    System.out.println(a);
    a += c; // not flagged
    System.out.println(a);
    // Errors printed out
    // d  = d + b;
    // d = d + c;
  }
}
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by oowekyala** [comment]
Hi, thanks for the suggestion, this is a good idea. PMD's type resolution is not yet precise enough to flag this very reliably, especially in java 10+ code with local variable type inference. This will get better with the next major release of PMD, and could be a new rule for that release.

**Comment by Fernal73** [comment]
Great. How about a Twitter account for PMD or is that to be a suggestion for Github (integration)?

**Comment by jsotuyod** [comment]
> Great. How about a Twitter account for PMD or is that to be a suggestion for Github (integration)?

There is a Twitter account for PMD: https://twitter.com/pmd_analyzer

We use it mainly for new releases announcements 'though.

**Comment by Fernal73** [comment]
Great. Thanks.
